infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -471,6 +470,21 @@ public void tokenRequest(CorfuPayloadMsg<TokenRequest> msg,
         }
     }
 
+    /**
+     * Service an incoming request to trim the sequencer.
+     */
+    @ServerHandler(type = CorfuMsgType.STREAMS_ADDRESS_TRIM)
+    public void streamsAddressSpaceTrim(CorfuPayloadMsg<StreamsAddressResponse> msg,
+                            ChannelHandlerContext ctx, IServerRouter r) {
+        Map<UUID, StreamAddressSpace> addressMap = msg.getPayload().getAddressMap();
+        addressMap.forEach(this::addressSpaceTrim);
+        r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
+    }
+
+    private void addressSpaceTrim(UUID streamId, StreamAddressSpace addressSpace) {
```

Comment:
add comments about why this function could be used even if it unsafe.
Rename it to addressSpaceTrimUnsafe